core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/FileAsyncRequestBody.java

```diff
@@ -271,34 +274,22 @@ private void closeFile() {
             }
         }
 
-        private void signalOnNext(ByteBuffer bb) {
-            synchronized (this) {
-                if (!done) {
-                    subscriber.onNext(bb);
-                }
-            }
-        }
-
         private void signalOnComplete() {
-            synchronized (this) {
```

Comment:
`synchronized` is removed because it doesn't seem to be necessary here imo.

Reply:
Can `signalOnComplete` and `signalOnError` be invoked at the same time though? It's either complete or error right?

Reply:
hmm, one (albeit rare possibility) if the file is complete, it's possible that the reading thread calls `onComplete()`, and at the same time, `request(-1)` is called so we try to signal `onError()`.

Reply:
+1 I believe this was needed to ensure mutual execution with signalOnError

Reply:
Yeah, good point 👍🏽